packages/aws-cdk-lib/core/lib/private/refs.ts

```diff
@@ -323,17 +337,23 @@ function createNestedStackOutput(producer: Stack, reference: Reference): CfnRefe
  *
  * Will create Outputs along the chain of Nested Stacks, and return the final `{ Fn::GetAtt }`.
  */
-export function referenceNestedStackValueInParent(reference: Reference, targetStack: Stack) {
+export function referenceNestedStackValueInParent(reference: Reference, targetStack: Stack): Intrinsic {
   let currentStack = Stack.of(reference.target);
   if (currentStack !== targetStack && !isNested(currentStack, targetStack)) {
     throw new Error(`Referenced resource must be in stack '${targetStack.node.path}', got '${reference.target.node.path}'`);
   }
 
+  const isNestedListReference = currentStack !== targetStack && reference.typeHint === ResolutionTypeHint.STRING_LIST;
+
   while (currentStack !== targetStack) {
     reference = createNestedStackOutput(Stack.of(reference.target), reference);
     currentStack = Stack.of(reference.target);
   }
 
+  if (isNestedListReference) {
+    return Tokenization.reverseList(Fn.split(STRING_LIST_REFERENCE_DELIMITER, Token.asString(reference))) as Intrinsic;
```

Comment:
The `as` operator pulls double duty here, since the return type is `IResolvable | undefined`. Might be safer to assert that the value is non-null before type casting it

Reply:
I was following the way that its done on lines 215-216 of this same file:

```
    // I happen to know this returns a Fn.split() which implements Intrinsic.
    return Tokenization.reverseList(importExpr) as Intrinsic;
```

Happy to address but do you have an example of the type of non-null assertion you'd use in this case?

Reply:
Just something I try to lookout for in general, but it doesn't seem to be a likely issue here. The `as` doesn't make it explicitly clear that the casting also acts as a non-null type assertion, so I usually like to separate them:

```ts
if (isNestedListReference) {
  const list = Tokenization.reverseList(Fn.split(STRING_LIST_REFERENCE_DELIMITER, Token.asString(reference)));
  if (!list) throw new Error('Parse error');
  return list as Intrinsic;
```

You could also use a guard function to do something similar, but it starts to become overkill